### ISSUE TITLE
translate-c error: no align available for type 'c_void'

### ISSUE THREAD

**Issue by dwlnetnl** [ziglang/zig#10222]
### Zig Version

0.9.0-dev.1737+c42763f8c

### Steps to Reproduce

When compiling the following Zig source importing a C header, there is a compiler error: `error: no align available for type 'c_void'`.

issue.zig:
```zig
const c = @cImport({
    @cInclude("header.h");
});

export fn onLoad(s: *c.someStruct) c_int {
    c.someFunc(s);
    return 0;
}
```

header.h:
```c
struct someStruct;
typedef struct someStruct someStruct;
static void someFunc(someStruct *s) {
    void *getapifuncptr = ((void **)s)[0];
}
```

### Expected Behavior

```
% ./zig-macos-x86_64-0.9.0-dev.1737+c42763f8c/zig build-lib -I. issue.zig
%
```

### Actual Behavior

```
% ./zig-macos-x86_64-0.9.0-dev.1737+c42763f8c/zig build-lib -I. issue.zig
./zig-macos-x86_64-0.9.0-dev.1737+c42763f8c/lib/zig/std/meta.zig:121:37: error: no align available for type 'c_void'
        .Optional => |info| switch (@typeInfo(info.child)) {
                                    ^
zig-cache/o/5157ecf9bfa15ebd540057475ceca27b/cimport.zig:54:98: note: called from here
    var getapifuncptr: ?*c_void = @ptrCast([*c]?*c_void, @alignCast(@import("std").meta.alignment(?*c_void), s))[@intCast(c_uint, @as(c_int, 0))];
                                                                                                 ^
zig-cache/o/5157ecf9bfa15ebd540057475ceca27b/cimport.zig:52:56: note: called from here
pub fn someFunc(arg_s: ?*someStruct) callconv(.C) void {
                                                       ^
./zig-macos-x86_64-0.9.0-dev.1737+c42763f8c/lib/zig/std/meta.zig:119:49: note: referenced here
pub fn alignment(comptime T: type) comptime_int {
                                                ^
./zig-macos-x86_64-0.9.0-dev.1737+c42763f8c/lib/zig/std/meta.zig:122:39: note: referenced here
            .Pointer, .Fn => alignment(info.child),
                                      ^
zig-cache/o/5157ecf9bfa15ebd540057475ceca27b/cimport.zig:54:98: note: referenced here
    var getapifuncptr: ?*c_void = @ptrCast([*c]?*c_void, @alignCast(@import("std").meta.alignment(?*c_void), s))[@intCast(c_uint, @as(c_int, 0))];
                                                                                                 ^
%
```

**Comment by dwlnetnl** [comment]
Is it a complicated bug to fix, if not I can have a crack at it.

**Comment by igaryhe** [comment]
hi,
i ran into the same problem. would like to know if it's going to be fixed or how should we fix it

**Comment by igaryhe** [comment]
the code causing the problem is this: https://github.com/nothings/stb/blob/master/stb_truetype.h#L2791-L2795

**Comment by jibal** [comment]
I don't know if it's related, but the compiler also produces "error: no align available for type 'comptime_int'" when calling `@typeInfo(@TypeOf(&x))` where `x` is a `comptime_int`. `@typeInfo(...)` should never fail regardless of what you throw at it.

**Comment by sskras** [comment]
Seems like the original issue was fixed some time before `0.11.0-dev.1594+a5d25fabd` (2023 Feb 10):

![image](https://github.com/ziglang/zig/assets/7887758/106cb6f9-cde3-4934-8291-3cbcc0463dbc)

At least on Windows.

**Comment by sskras** [comment]
On Linux it's fixed too, the version `0.11.0-dev.4002+7dd1cf26f`:

![image](https://github.com/ziglang/zig/assets/7887758/76cdedb5-a41d-4b93-b613-78cbe47340d8)

In case no one is going to identify the fixing commit id, the issue can be closed, I guess.
